tests/admin_filters/tests.py

```diff
@@ -1315,6 +1337,390 @@ def test_simplelistfilter_with_queryset_based_lookups(self):
         self.assertIs(choices[2]["selected"], False)
         self.assertEqual(choices[2]["query_string"], "?publication-decade=the+00s")
 
+    def test_filters_with_facets(self):
+        modeladmin = DecadeFilterBookAdmin(Book, site)
+
+        # Facets with no filters applied
+        request = self.request_factory.get("/?_facets")
+        request.user = self.alfred
+        changelist = modeladmin.get_changelist_instance(request)
+        queryset = changelist.get_queryset(request)
+        self.assertSequenceEqual(list(queryset), list(Book.objects.order_by("-id")))
+
+        filterspec_relatedfieldlistfilter = changelist.get_filters(request)[0][0]
+        choices = list(filterspec_relatedfieldlistfilter.choices(changelist))
+        self.assertEqual(choices[1]["display"], "alfred (2)")
+        self.assertEqual(choices[2]["display"], "bob (1)")
+        for choice in choices:
+            self.assertIn("_facets", choice["query_string"])
+
+        filterspec_simplelistfilter = changelist.get_filters(request)[0][1]
+        choices = list(filterspec_simplelistfilter.choices(changelist))
+        self.assertEqual(choices[1]["display"], "the 2000's (2)")
+        self.assertEqual(choices[2]["display"], "the 1990's (1)")
+        self.assertEqual(choices[3]["display"], "the 1980's (0)")
+        self.assertEqual(choices[4]["display"], "other decades (-)")
```

Comment:
Can we add a hook for checking `choices`? For example:
```python
def assertChoicesDisplay(self, choices, expected_displays):
    for choice, expected_display in zip(choices[1:], expected_displays, strict=True):
        self.assertEqual(choice["display"], expected_display])
```
and use it as:
```python
    self.assertChoicesDisplay(
        choices, ["the 2000's (2)", "the 1990's (1)", "the 1980's (0)", "other decades (-)"]
    )
```

Reply:
Nice idea :+1: